### ISSUE TITLE
size mismatch issue

### ISSUE THREAD

**Issue by aquorio15** [chenhaoxing/DeMamba#1]
Hi, First of all thanks for making the repository public, I was going through your code seems to me there are some issues with the code for example. In line no. 35 of the DeMamba.py
```
_, _, c = sequence_output.shape
```
The output from CLIP is a 2D tensor consisting of (batch_size, logits)
Secondly in line no. 145-149 I am getting a shape mismatch error, it does not matter what inputs I choose
I have used an input of torch.randn(1, 16, 3, 224, 224) to check the model

Thank you, any kind of help would be greatly appreciated

**Comment by chenhaoxing** [comment]
Hi, we have modified the output of the CLIP, as referenced in https://github.com/chenhaoxing/DeMamba/blob/main/models/clip/model.py. The normal CLIP output is 
```
x = self.ln_post(x[:, :1, :]),
```
which yields the CLS token of the ViT. To obtain spatial information, we have altered this to 
```
x = self.ln_post(x[:, 1:, :]).
```
